File: kuark-base/src/org/kuark/base/support/ICallback.java
package org.kuark.base.support;

import java.io.Serializable;

/**
 * 回调接口
 *
 * @since 1.0.0
 */
public interface ICallback<P, R> extends Serializable{

    /**
     * 回调行为
     *
     * @param p 参数
     * @return 返回值
     */
    R execute(P p);

}

File: kuark-base/src/io/kuark/base/support/ICallback.kt
package io.kuark.base.support

import java.io.Serializable

/**
 * 回调接口
 *
 * @param P 参数类型
 * @param R 返回值类型
 * @author K
 * @since 1.0.0
 */
fun interface ICallback<P, R> : Serializable {

    /**
     * 回调行为
     *
     * @param p 参数
     * @return 返回值
     * @author K
     * @since 1.0.0
     */
    fun execute(p: P): R

}